menas/ui/service/RuleService.js

```diff
@@ -52,10 +52,12 @@ var RuleService = new function () {
   };
 
   this.removeRule = function (oCurrentDataset, iRuleIndex) {
-    let conformance = oCurrentDataset["conformance"].filter((_, index) => index !== iRuleIndex);
-    let newDataset = {...oCurrentDataset, conformance: conformance};
+    let conformance = oCurrentDataset["conformance"]
+      .filter((_, index) => index !== iRuleIndex)
+      .sort((first, second) => first.order > second.order)
+      .map((currElement, index) =>  {return {...currElement, order: index}});
 
-    return newDataset;
+    return {...oCurrentDataset, conformance: conformance};
```

Comment:
Don't know enough to make a judgement, just curious. Couldn't this be done like?
```
oCurrentDataset.conformance = conformance;
return oCurrentDataset;
```

Reply:
This will create new object with old values except conformance.
It is more functional approach than mutate state of object. 
So yes it could be done like your suggestion. 
Should I change it ?

Reply:
Cool man, I like your approach!